pkg/strategy/streaming.go

```diff
@@ -15,84 +17,235 @@ import (
 	"github.com/jaegertracing/jaeger-operator/pkg/deployment"
 	"github.com/jaegertracing/jaeger-operator/pkg/ingress"
 	"github.com/jaegertracing/jaeger-operator/pkg/inject"
+	"github.com/jaegertracing/jaeger-operator/pkg/kafka"
 	"github.com/jaegertracing/jaeger-operator/pkg/route"
 	"github.com/jaegertracing/jaeger-operator/pkg/storage"
 )
 
 func newStreamingStrategy(jaeger *v1.Jaeger) S {
-	c := S{typ: Streaming}
+	manifest := S{typ: Streaming}
 	collector := deployment.NewCollector(jaeger)
 	query := deployment.NewQuery(jaeger)
 	agent := deployment.NewAgent(jaeger)
 	ingester := deployment.NewIngester(jaeger)
 
 	// add all service accounts
 	for _, acc := range account.Get(jaeger) {
-		c.accounts = append(c.accounts, *acc)
+		manifest.accounts = append(manifest.accounts, *acc)
 	}
 
 	// add all cluster role bindings
-	c.clusterRoleBindings = crb.Get(jaeger)
+	manifest.clusterRoleBindings = crb.Get(jaeger)
 
 	// add the config map
 	if cm := configmap.NewUIConfig(jaeger).Get(); cm != nil {
-		c.configMaps = append(c.configMaps, *cm)
+		manifest.configMaps = append(manifest.configMaps, *cm)
 	}
 
 	// add the Sampling config map
 	if cm := sampling.NewConfig(jaeger).Get(); cm != nil {
-		c.configMaps = append(c.configMaps, *cm)
+		manifest.configMaps = append(manifest.configMaps, *cm)
+	}
+
+	_, pfound := asOptionsMap(jaeger.Spec.Collector.Options.Map())["kafka.producer.brokers"]
+	_, cfound := asOptionsMap(jaeger.Spec.Ingester.Options.Map())["kafka.consumer.brokers"]
+	provisioned := jaeger.Annotations[v1.AnnotationProvisionedKafkaKey] == v1.AnnotationProvisionedKafkaValue
```

Comment:
The comment associated with these constants implies that the annotation is to go on `Kafkas` - i.e. I assume this means the kafka CR/deployment? Not sure what its role is on the jaeger CR, seeing as the key is not kafka specific?

Reply:
That's why it's good to have reviews :) At first, I thought about adding this annotation to the managed resources, but I actually needed a hint in the Jaeger CR itself. 

On the first reconciliation, we determine whether or not to provision a Kafka by the absence of the Broker URLs in the configuration. Once we provision the cluster, the Streaming strategy sets the Collector's and Ingester's options to point to the provisioned cluster.

On the second reconciliation, the brokers list is filled, so, we can't know if the cluster was provisioned by us, hence the need for the hint.

So, the correct approach is to make the annotation specific to Kafka.